src/plugins/discover/public/application/angular/directives/no_results.tsx

```diff
@@ -0,0 +1,235 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { Fragment } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiCode,
+  EuiDescriptionList,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { getServices } from '../../../kibana_services';
+import { DataPublicPluginStart } from '../../../../../data/public';
+
+export function DiscoverNoResults({
+  timeFieldName,
+  queryLanguage,
+  error,
+  data,
+}: {
+  timeFieldName: string;
+  queryLanguage: string;
+  error: Error;
+  data: DataPublicPluginStart;
+}) {
+  let timeFieldMessage;
```

Comment:
Could we have a function instead that returns `timeFieldMessage` if proper params are set and null otherwise? I think it would look a lot nicer.